compiler/rustc_hir_typeck/src/pat.rs

```diff
@@ -2385,11 +2385,10 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         min_len: u64,
     ) -> (Option<Ty<'tcx>>, Ty<'tcx>) {
         let len = match len.eval(self.tcx, self.param_env, span) {
-            // FIXME(BoxyUwU): Assert the `Ty` is a `usize`?
```

Comment:
This assert hasn't been added

Reply:
`to_target_usize` is implicitly asserting that